src/pages/index.js

```diff
@@ -146,22 +168,34 @@ export function updateProfile(profileObject) {
       name: newName,
       occupation: newOccupation,
     })
+    .then(() => {
+      userInfo.setUserInfo(newName, newOccupation);
+      editProfileFormValidator.disableSubmitButton();
+      editProfileModal.close();
+    })
     .finally(() => {
       profileSubmitButton.textContent = "Save";
+    })
+    .catch((error) => {
+      console.error("Oops, something happened", error);
     });
-  userInfo.setUserInfo(newName, newOccupation);
-  editProfileFormValidator.disableSubmitButton();
-  editProfileModal.close();
 }
 
 export function updateProfilePicture(profileObject) {
   const newPhotoLink = profileObject["input-url"];
   console.log(profileObject);
   changeAvatarSubmitButton.textContent = "Saving...";
-  api.updateAvatar(newPhotoLink).then((response) => {
-    userInfo.setUserAvatar(response.avatar);
-    changeAvatarSubmitButton.textContent = "Save";
-  });
+  api
+    .updateAvatar(newPhotoLink)
+    .then((response) => {
+      userInfo.setUserAvatar(response.avatar);
+    })
+    .finally(() => {
+      changeAvatarSubmitButton.textContent = "Save";
+    })
+    .catch((error) => {
+      console.error("Oops, something happened", error);
+    });
   updateAvatarFormValidator.disableSubmitButton();
   updateAvatarModal.close();
```

Comment:
* You haven't corrected my remark in the other places of the code https://github.com/angelbaezsa/se_project_aroundtheus/pull/3#discussion_r1255471923 
* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well